apiserver/pkg/printers/projectcalico/printers.go

```diff
@@ -0,0 +1,169 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package projectcalico
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/duration"
+	"k8s.io/kubernetes/pkg/printers"
+
+	calico "github.com/projectcalico/api/pkg/apis/projectcalico/v3"
+)
+
+// AddHandlers adds print handlers for projectcalico types.
+func AddHandlers(h printers.PrintHandler) {
+	calicoNodeStatusColumnDefinitions := []metav1.TableColumnDefinition{
+		{Name: "Name", Type: "string", Format: "name", Description: metav1.ObjectMeta{}.SwaggerDoc()["name"]},
+		{Name: "Node", Type: "string", Priority: 0, Description: "The node name identifies the Calico node instance for node status."},
+		{Name: "Classes", Type: "string", Description: "Types of information to monitor for this calico/node."},
+		{Name: "Update Interval", Type: "string", Description: "Period in seconds at which CalicoNodeStatus should be updated."},
+		{Name: "Age", Type: "string", Description: metav1.ObjectMeta{}.SwaggerDoc()["creationTimestamp"]},
+		{Name: "last Updated", Type: "string", Description: "Last time the status has been updated"},
+		{Name: "Agent State", Type: "string", Priority: 1, Description: "The state of the BGP Daemon."},
+		{Name: "BGP Peers", Type: "string", Priority: 1, Description: "Number of BGP Peers in the format of total/established/non-established."},
+		{Name: "BGP Routes", Type: "string", Priority: 1, Description: "Number of routes learned from BGP peers."},
+	}
+	h.TableHandler(calicoNodeStatusColumnDefinitions, printCalicoNodeStatusList)
+	h.TableHandler(calicoNodeStatusColumnDefinitions, printCalicoNodeStatus)
+}
+
+func printCalicoNodeStatusList(statusList *calico.CalicoNodeStatusList, options printers.GenerateOptions) ([]metav1.TableRow, error) {
+	rows := make([]metav1.TableRow, 0, len(statusList.Items))
+	for i := range statusList.Items {
+		r, err := printCalicoNodeStatus(&statusList.Items[i], options)
+		if err != nil {
+			return nil, err
+		}
+		rows = append(rows, r...)
+	}
+	return rows, nil
+}
+
+func printCalicoNodeStatus(status *calico.CalicoNodeStatus, options printers.GenerateOptions) ([]metav1.TableRow, error) {
+	row := metav1.TableRow{
+		Object: runtime.RawExtension{Object: status},
+	}
+
+	// Define a function return true if the status contains information about a class.
+	hasClass := func(c calico.NodeStatusClassType) bool {
+		for _, class := range status.Spec.Classes {
+			if class == c {
+				return true
+			}
+		}
+		return false
+	}
+
+	var classes string
+	for _, class := range status.Spec.Classes {
+		classes += fmt.Sprintf("%s,", class)
+	}
+	classesStr := fmt.Sprintf("[%s]", strings.TrimSuffix(classes, ","))
```

Comment:
I'd probably format this without the brackets and just do comma separated - e.g.,

"Agent,Routes"

I don't think the brackets make it more readable, so fewer characters is better.